### PR TITLE
error in line 23 i.e a constant value

### PR DESCRIPTION
**Pull request by faizanahamed1414**
in line 23.
```python
int a[n];
```
give error while running i.e. expression must have a constant value.
so, rectifying this we can use pointer.
```python 
int* a = new int[n];
```
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTION.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [ ] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTION.md#Commit-Guidelines)
- [ ] Search previous suggestions before making a new one, as yours may be a duplicate.
- [ ] Sort by alphabetical order
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Comment by cclauss** [comment]
Please click on the __Details__ link next to the ❌ to see why the tests are failing.  Our linter does not like tab indentation.

**Comment by cclauss** [comment]
Please use `std::endl` instead of `endl`.

**Comment by faizanahamed1414** [comment]
Following changes made:
1) Replace of "int a[n]" by "int* a = new int[n];" in line 19 
2) change file name from "Binary Search" to binary_search.
3) add "std::"
4) remove tab
Purpose of changes:
1) give error while running.
2) test fail occurs while request for pull

File: search/Binary Search.cpp [hunk 1]
```diff
@@ -11,7 +11,7 @@ l = m + 1;
 }
 return -1;
 }
-int main(int argc, char const *argv[]) {
+int main() {
```
Comment (cclauss) [inline search/Binary Search.cpp]: Why remove?

Comment (faizanahamed1414) [reply search/Binary Search.cpp]: I again add this in last commit before that I did not able to find argc variable and thought might be it give error.
Then I again add this in new commit.

**Pull request merged**